autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding.py

```diff
@@ -0,0 +1,108 @@
+from typing import Dict, Optional, Tuple, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+import ConfigSpace.hyperparameters as CSH
+from ConfigSpace import EqualsCondition
+
+import numpy as np
+import pandas as pd
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class BagOfWordEncoder(AutoSklearnPreprocessingAlgorithm):
+    def __init__(
+        self,
+        ngram_range: Optional[int] = None,
+        min_df_choice: Optional[str] = None,
+        min_df_absolute: Optional[int] = None,
+        min_df_relative: Optional[float] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.ngram_range = ngram_range
+        self.random_state = random_state
+        self.min_df_choice = min_df_choice
+        self.min_df_absolute = min_df_absolute
+        self.min_df_relative = min_df_relative
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'BagOfWordEncoder':
+
+        if isinstance(X, pd.DataFrame):
+            # define a CountVectorizer for every feature (implicitly defined by order of columns,
+            # maybe change the list
+            # to a dictionary with features as keys)
+            if self.min_df_choice == "min_df_absolute":
+                self.preprocessor = CountVectorizer(min_df=self.min_df_absolute,
+                                                    ngram_range=(1, self.ngram_range))
+            elif self.min_df_choice == "min_df_relative":
+                self.preprocessor = CountVectorizer(min_df=self.min_df_relative,
+                                                    ngram_range=(1, self.ngram_range))
+            else:
+                raise KeyError()
+            for feature in X.columns:
+                self.preprocessor = self.preprocessor.fit(X[feature])
+        else:
+            raise ValueError("Your text data is not encoded in a pandas.DataFrame\n"
+                             "Please make sure to use a pandas.DataFrame and ensure"
+                             "that the text features are encoded as strings.")
+        return self
+
+    def transform(self, X: PIPELINE_DATA_DTYPE) -> PIPELINE_DATA_DTYPE:
+        X_new = None
+        if self.preprocessor is None:
+            raise NotImplementedError()
+        # iterate over the pretrained preprocessors and columns and transform the data
+        for feature in X.columns:
+            if X_new is None:
+                X_new = self.preprocessor.transform(X[feature])
+            else:
+                X_new += self.preprocessor.transform(X[feature])
+        return X_new
```

Comment:
If I understand this correctly, this would transform a pandas dataframe with string 3 columns and return just one column with their sum?